File: 34-一品威客登录-AES + MD5/param.py
import requests, execjs, time

def ex_js(filepath, func_name, *args):
    with open(filepath, 'r', encoding='utf-8') as f:
        js_code = f.read()
    js_code = execjs.compile(js_code)
    return js_code.call(func_name, *args)


def login_part():
    login_url = 'https://www.epwk.com/api/epwk/v1/user/login'
    cookies = {
        'HWWAFSESID': '50f5773c86cfc078bc',
        'HWWAFSESTIME': '1758430600928',
        'Hm_lvt_387b8f4fdb89d4ea233922bdc6466394': '1758363320,1758430601',
        'Hm_lpvt_387b8f4fdb89d4ea233922bdc6466394': '1758430601',
        'HMACCOUNT': '5B53905BB2005420',
        'PHPSESSID': '27ca7539e33e2a21332ed39b07e0db74bc703089',
        'time_diff': '-1',
        'XDEBUG_SESSION': 'XDEBUG_ECLIPSE',
        'banners_show_cookie_ip': '118.124.67.16',
        'login_fail_need_graphics': '1',
    }
    headers = {
        'Accept': 'application/json, text/plain, */*',
        'Accept-Language': 'zh-CN,zh;q=0.9',
        'Access-Token': '',
        'App-Id': '4ac490420ac63db4',
        'App-Ver': '',
        'CHOST': 'www.epwk.com',
        'Cache-Control': 'no-cache',
        'Connection': 'keep-alive',
        'Content-Type': 'application/x-www-form-urlencoded',
        'Device-Os': 'web',
        'Device-Ver': '',
        'Imei': '',
        'NonceStr': '17584307606hyak',
        'Origin': 'https://www.epwk.com',
        'Os-Ver': '',
        'Pragma': 'no-cache',
        'Referer': 'https://www.epwk.com/login.html',
        'Sec-Fetch-Dest': 'empty',
        'Sec-Fetch-Mode': 'cors',
        'Sec-Fetch-Site': 'same-origin',
        'Signature': 'qQ8DTYCWRq3OgKgzT4HURNz+9GNhnbTzhRXJ/pUavP06vci9VoDuVJ0PSzx7QFbV',
        'Timestemp': '1758430760',
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/140.0.0.0 Safari/537.36',
        'X-REQUEST-ID': '192821d85cae749d070a1971a87c827c',
        'sec-ch-ua': '"Chromium";v="140", "Not=A?Brand";v="24", "Google Chrome";v="140"',
        'sec-ch-ua-mobile': '?0',
        'sec-ch-ua-platform': '"Windows"',
        # 'Cookie': 'HWWAFSESID=50f5773c86cfc078bc; HWWAFSESTIME=1758430600928; Hm_lvt_387b8f4fdb89d4ea233922bdc6466394=1758363320,1758430601; Hm_lpvt_387b8f4fdb89d4ea233922bdc6466394=1758430601; HMACCOUNT=5B53905BB2005420; PHPSESSID=27ca7539e33e2a21332ed39b07e0db74bc703089; time_diff=-1; XDEBUG_SESSION=XDEBUG_ECLIPSE; banners_show_cookie_ip=118.124.67.16; login_fail_need_graphics=1',
    }
    data = {
        'username': '15712345678',
        'password': '123456678',
        'code': 'r3qz',
        'hdn_refer': '',
    }

    response = requests.post(login_url, cookies=cookies, headers=headers, data=data)

    return response.text

File: 34-一品威客登录-AES + MD5/param.js
const CryptoJS = require('crypto-js');

const l = {
    key: CryptoJS.enc.Utf8.parse("fX@VyCQVvpdj8RCa"),
    iv: CryptoJS.enc.Utf8.parse(function (t) {
        for (var e = "", i = 0; i < t.length; i += 2) {
            var n = parseInt(t[i] + "" + t[i + 1], 16);
            e += String.fromCharCode(n);
        }
        return e;
    }("00000000000000000000000000000000"))
};

function get_app() {
    var e = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "",
        t = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {
            fromRes: !1, parseJSON: true
        }
    // 根据实际需求添加返回值
    return null; // 或者其他适当的返回值
}

function uuid() {
    let e = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : 5
    return Math.random().toString(36).substring(3, 3 + e)
}

function get_app() {
    var e = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "",
        t = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {
            fromRes: !1, parseJSON: true
        }
}

function d(data) {
    return CryptoJS.MD5(data).toString()
}

function v(data) {

    return function (data) {
        return CryptoJS.AES.encrypt(data, l.key, {
            iv: l.iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        }).toString()
    }(data)
}

function aesSign(t) {
    var data = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {},
        e = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : "a75846eb4ac490420ac63db46d2a03bf",
        n = e + f(data) + f(t) + e;
    return n = d(n),
        n = v(n)
}

function f(t) {
    let e = "";
    // 使用 reduce 替代 forEach 以便返回结果
    return Object.keys(t).sort().reduce(function (acc, n) {
        // 补全括号，修正三元运算符
        acc += n + ("object" === typeof (t[n])
                ? JSON.stringify(t[n], function (t, e) {
                    return "number" == typeof e ? String(e) : e;
                }).replace(/\//g, "\\/")  // 补全缺失的括号
                : t[n]
        );
        return acc;
    }, e);  // 初始值为 e
}

function get_signature() {
    let timestemp = parseInt(new Date().getTime() / 1000)
    const COM_TIME_DIFF = get_app('time_diff') || 0
    timestemp = timestemp - COM_TIME_DIFF
    const Header = {
        'App-Ver': '',
        'Os-Ver': '',
        'Device-Ver': '',
        Imei: '',
        'Access-Token': '',
        Timestemp: timestemp,
        NonceStr: `${timestemp}${uuid()}`,
        'App-Id': "4ac490420ac63db4",
        'Device-Os': 'web'
        // 'User-Agent': 'web'
    }
    let signatureParams = {
        "username": "12324355646", "password": "sadsadsaf", "code": "dypn", "hdn_refer": ""
    }

    return aesSign(Header, signatureParams, "4ac490420ac63db4")
}


function get_signature_code() {
    let timestemp = parseInt(new Date().getTime() / 1000)
    const COM_TIME_DIFF = get_app('time_diff') || 0
    timestemp = timestemp - COM_TIME_DIFF
    const Header = {
        'App-Ver': '',
        'Os-Ver': '',
        'Device-Ver': '',
        Imei: '',
        'Access-Token': '',
        Timestemp: timestemp,
        NonceStr: `${timestemp}${uuid()}`,
        'App-Id': "4ac490420ac63db4",
        'Device-Os': 'web'
        // 'User-Agent': 'web'
    }
    let signatureParams = {}

    return aesSign(Header, signatureParams, "4ac490420ac63db4")
    
}

console.log(get_signature_code())
console.log(get_signature())
